### PR TITLE
HL-94

### PR DESCRIPTION
**Pull request by OptimusOpus**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-94)

Add deployments for MarketOracle and Registry contracts

### CODE REVIEW

**Review by JasminFarhi — APPROVED** [review]
Not super familiar but LGTM

File: scripts/deploy/contracts/MarketOracle.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export const deploy = async (deployer, setAddresses) => {
+  console.log("deploying marketOracle");
+  const signer = deployer ?? (await getSignerForDeployer());
+  const factory = new MarketOracle__factory(signer);
+  const args = constructorArguments();
+  const contract = await factory.deploy(...args);
+  await contract.deployTransaction.wait(1);
+  console.log(`deployed marketOracle to address ${contract.address}`);
+  setAddresses({ marketOracle: contract.address });
+  //await verifyOnEtherscan(contract.address, args);
```
Comment (brettvniekerk) [inline scripts/deploy/contracts/MarketOracle.ts:24]: why is this commented?

File: scripts/deploy/contracts/MarketOracle.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  return contract as Contract;
```
Comment (brettvniekerk) [inline scripts/deploy/contracts/MarketOracle.ts:25]: is this typecast needed?

File: scripts/deploy/contracts/Registry.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export const deploy = async (deployer, setAddresses) => {
+  console.log("deploying registry");
+  const signer = deployer ?? (await getSignerForDeployer());
+  const factory = new Registry__factory(signer);
+  const args = constructorArguments();
+  const contract = await factory.deploy(...args);
+  await contract.deployTransaction.wait(5);
+  console.log(`deployed registry to address ${contract.address}`);
+  setAddresses({ registry: contract.address });
+  //await verifyOnEtherscan(contract.address, args);
```
Comment (brettvniekerk) [inline scripts/deploy/contracts/Registry.ts:27]: same here

File: scripts/deploy/contracts/Registry.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  return contract as Contract;
```
Comment (brettvniekerk) [inline scripts/deploy/contracts/Registry.ts:28]: same here

**Review by bazmatic — COMMENTED** [review]
LGTM. This code will be redundant when HL-69 is merged anyway so I will refrain from fussing on details.

File: scripts/deploy/contracts/MarketOracle.ts (hunk 2 above)
Comment (OptimusOpus) [reply scripts/deploy/contracts/MarketOracle.ts:25]: Yeah, the index file that imports this was upset when I removed it

**Review by bazmatic — APPROVED** [review]

File: scripts/deploy/contracts/MarketOracle.ts (hunk 1 above)
Comment (OptimusOpus) [reply scripts/deploy/contracts/MarketOracle.ts:24]: Barry is working on the deployment scripts, this PR is just for deployment scripts for the contracts I needed that didn't have them

**Pull request merged**